torch/_inductor/codegen/common.py

```diff
@@ -329,30 +335,40 @@ def cpp_argdefs(self):
             {name: val.dtype for name, val in V.graph.constants.items()}
         )
 
+        lhs_idx_type, rhs_idx_type = self.get_idx_type()
+
         call_args = []
         arg_defs = []
+        arg_types = []
         for inplaced in unique(self.inplace_buffers.values()):
             outer = inplaced.other_names[-1]
             inner = inplaced.inner_name
             dtype = buffer_types[outer]
+            ptr_type = self.get_ptr_type(dtype)
             arg_defs.append(f"{DTYPE_TO_CPP[dtype]}* __restrict__ {inner}")
-            call_args.append(f"c_void_p({outer}.data_ptr())")
+            call_args.append(f"{ptr_type}({outer}.data_ptr())")
+            arg_types.append(f"{DTYPE_TO_CPP[dtype]}*")
         for outer, inner in self.input_buffers.items():
             if outer in self.inplace_buffers:
                 continue
             dtype = buffer_types[outer]
+            ptr_type = self.get_ptr_type(dtype)
             arg_defs.append(f"const {DTYPE_TO_CPP[dtype]}* __restrict__ {inner}")
-            call_args.append(f"c_void_p({outer}.data_ptr())")
+            call_args.append(f"{ptr_type}({outer}.data_ptr())")
+            arg_types.append(f"const {DTYPE_TO_CPP[dtype]}*")
         for outer, inner in self.output_buffers.items():
             if outer in self.inplace_buffers or inner == "REMOVED":
                 continue
             dtype = buffer_types[outer]
+            ptr_type = self.get_ptr_type(dtype)
             arg_defs.append(f"{DTYPE_TO_CPP[dtype]}* __restrict__ {inner}")
-            call_args.append(f"c_void_p({outer}.data_ptr())")
+            call_args.append(f"{ptr_type}({outer}.data_ptr())")
+            arg_types.append(f"{DTYPE_TO_CPP[dtype]}*")
         for outer, inner in self.sizevars.items():
             arg_defs.append(f"const {INDEX_TYPE} {inner}")
-            call_args.append(f"c_long({outer})")
-        return arg_defs, call_args
+            call_args.append(f"{lhs_idx_type}{outer}{rhs_idx_type}")
```

Comment:
This might be cleaner if you added a helper function like:
```suggestion
            call_args.append(self.wrap_size_arg(outer))
```

Reply:
I've added helper functions for this part.